addons/web_unsplash/controllers/main.py

```diff
@@ -89,22 +89,18 @@ def slugify(s):
                     continue
 
                 # get mime-type of image url because unsplash url dosn't contains mime-types in url
-                mimetype = req.headers.get('Content-Type')
-                datas = req.content
+                image_base64 = base64.b64encode(req.content)
             except requests.exceptions.ConnectionError as e:
                 logger.exception("Connection Error: " + str(e))
                 continue
             except requests.exceptions.Timeout as e:
                 logger.exception("Timeout: " + str(e))
                 continue
 
-            # optimized image before save
-            if mimetype in ('image/jpeg', 'image/png'):
-                image = Image.open(io.BytesIO(datas))
-                if image.format in ('PNG', 'JPEG'):
-                    datas = tools.image_save_for_web(image)
-                    # append image extension in name
-                    query += '.' + str.lower(image.format)
+            image_base64 = tools.image_process(image_base64)
```

Comment:
That unsplash code (double mimetype check) was subject to debate and done for a particular reason but I can't remember why, I should have commented the code, we will see what happen :yum:.

Reply:
The new method is making sure the received file is a correct image and the mimetype is guessed accordingly too, so hopefully the change won't break anything.